workspace/viewer/PdfJsViewer.ts

```diff
@@ -515,49 +537,55 @@ export class PdfJsViewer {
 
   onResize() {
     if (this.currentFile) {
-      this.openCurrentPage();
+      this.showPage(this.currentPage);
     }
   }
 
   pageUp = () => {
     if (this.currentPage > 1) {
       this.currentPage -= 1;
-      this.openCurrentPage();
+      this.showPage(this.currentPage);
     }
   };
 
   pageDown = () => {
     if (this.currentFile && this.currentPage < this.currentFile.numPages) {
       this.currentPage += 1;
-      this.openCurrentPage();
+      this.showPage(this.currentPage);
     }
   };
 
-  getCurrentPage(): number | undefined { return this.currentPage; }
+  getCurrentPage(): number | undefined {
+    return this.currentPage;
+  }
 
   getZoomScale(): number { return this.zoom.current(); }
 
   // Toolbar button handlers
   zoomIn = () => {
-    this.zoom.zoomIn() && this.openCurrentPage();
+    if (this.currentFileUrl){
```

Comment:
This probably works for now but please add TODO to make it more delicate. It currently just reloads all the pages with new zoom scale which is probably not necessary.

Reply:
Done.
Now it updates the scale by calling `update(scale)` and draws(method `draw()`) page using new scale.